src/main/java/in/org/projecteka/hiu/dataflow/HealthInfoManager.java

```diff
@@ -27,11 +28,11 @@
 
     private Flux<DataEntry> getDataEntries(String transactionId, String hipId, String hipName) {
         return healthInformationRepository.getHealthInformation(transactionId)
-                .map(data -> DataEntry.builder()
+                .map(healthInfo -> DataEntry.builder()
                         .hipId(hipId)
                         .hipName(hipName)
-                        //.status(data != null ? Status.COMPLETED : Status.REQUESTED)
-                        .data(data)
+                        .status(EntryStatus.valueOf((String) healthInfo.get("status")))
```

Comment:
What happens if `healthInfo.get("status")` is not one of the possible values?

Reply:
there ideally would never be a scenario of the status field not being present or being invalid. but ok, can be better checked